File: src/geo-backend/geo_lib/websocket/modules/delete_job_module.py
"""
Delete job WebSocket module.
"""

import logging

from geo_lib.websocket.base_module import BaseWebSocketModule

logger = logging.getLogger(__name__)


class DeleteJobModule(BaseWebSocketModule):
    """WebSocket module for delete job functionality."""

    @property
    def module_name(self) -> str:
        return "delete_job"

    async def handle_message(self, message_type: str, data: dict) -> None:
        """Handle incoming messages for delete job module."""
        if message_type == 'refresh':
            await self.send_initial_state()
        else:
            logger.warning(f"Unknown message type for delete_job module: {message_type}")

    async def send_initial_state(self) -> None:
        """Send initial state for delete job module."""
        # Delete jobs don't have persistent state, so send empty state
        await self.send_to_client('initial_state', {})

    # Delete job event handlers
    async def started(self, event):
        """Handle delete_job_started event."""
        await self.send_to_client('started', event['data'])

    async def status_updated(self, event):
        """Handle delete_job_status_updated event."""
        await self.send_to_client('status_updated', event['data'])

    async def completed(self, event):
        """Handle delete_job_completed event."""
        await self.send_to_client('completed', event['data'])

    async def failed(self, event):
        """Handle delete_job_failed event."""
        await self.send_to_client('failed', event['data'])

File: src/frontend/src/assets/js/websocket/modules/DeleteJobModule.js
/**
 * Delete job WebSocket module for handling delete job events.
 */

import { BaseModule } from './BaseModule.js';

export class DeleteJobModule extends BaseModule {
    constructor(store) {
        super(store);
        this.moduleName = 'delete_job';
    }

    /**
     * Initialize the delete job module
     */
    initialize() {
        super.initialize();

        // Handle initial state
        this.subscribe('initial_state', (data) => {
            // Delete jobs don't have persistent state, so no action needed
        });

        // Handle delete job started - update item state
        this.subscribe('started', (data) => {
            // Mark item as deleting in the table
            this.store.dispatch('updateImportTableItem', {
                id: data.item_id,
                updates: { deleting: true }
            });
        });

        // Handle delete job status updated
        this.subscribe('status_updated', (data) => {
            // Could update progress here if needed
            this.store.dispatch('updateImportTableItem', {
                id: data.item_id,
                updates: { 
                    deleting: true,
                    deleteProgress: data.progress 
                }
            });
        });

        // Handle delete job completed - remove item
        this.subscribe('completed', (data) => {
            // Remove the deleted item from the table
            this.store.dispatch('removeImportTableItem', data.item_id);
        });

        // Handle delete job failed - clear deleting state
        this.subscribe('failed', (data) => {
            // Clear deleting state and optionally set error
            this.store.dispatch('updateImportTableItem', {
                id: data.item_id,
                updates: { 
                    deleting: false,
                    deleteError: data.error 
                }
            });
        });
    }
}
